### PR TITLE
Fix metric logging

### PR DESCRIPTION
**Pull request by maanug-nv**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Remove `rank_zero_only=True` flag, unless used as [documented](https://lightning.ai/docs/pytorch/stable/visualize/logging_advanced.html#rank-zero-only)
- Refactor train and val loss logging

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/lightning/_strategy_lib.py [hunk 1]
```diff
@@ -524,3 +524,23 @@ def load_model_state_dict(megatron_parallel, checkpoint: Mapping[str, Any], stri
...
         module.load_state_dict(_state_dict, strict=strict)
+
+
+def _sync_from_last_pipeline_stage(value: torch.Tensor, broadcast: bool = False):
+    from megatron.core import parallel_state
+
+    if parallel_state.get_pipeline_model_parallel_world_size() > 1:
+        src_rank = parallel_state.get_pipeline_model_parallel_last_rank()
+
+        if not broadcast:
```
Comment (ShriyaPalsamudram) [inline nemo/lightning/_strategy_lib.py:535]: Add a comment to explain exactly why this mode exists and what it is trying to achieve

File: nemo/lightning/pytorch/strategies.py [hunk 2]
```diff
@@ -479,7 +469,23 @@ def validation_step(self, dataloader_iter, *args: Any, **kwargs: Any) -> STEP_OU
 
         with self.precision_plugin.val_step_context():  # TODO: Do we need this?
             out = self.model(dataloader_iter, forward_only=True, *args, **kwargs)
-            self.lightning_module.log('val_loss', out, rank_zero_only=True, batch_size=1)
+
+            from megatron.core import parallel_state
+
+            pp_size = parallel_state.get_pipeline_model_parallel_world_size()
+            if pp_size > 1:
+                # ranks that are not final pp stage have 0 for loss, and out will be reduced
```
Comment (ericharper) [inline nemo/lightning/pytorch/strategies.py:477]: let's expand on this comment some more

**Review by marcromeyn — APPROVED** [review]

**Comment by ko3n1g** [comment]
stopping this until NeMo CI is out of maintenance mode
